app/com/mohiva/play/compressor/CompressorFilter.scala

```diff
@@ -88,17 +88,30 @@ abstract class CompressorFilter[C <: Compressor] extends Filter {
 
     if (isCompressible(result)) {
       result.body match {
-        case HttpEntity.Strict(data, contentType) =>
-          Future.successful(Result(result.header, HttpEntity.Strict(ByteString(compress(data)), contentType)))
-        case HttpEntity.Streamed(data, _, contentType) =>
-          data.toMat(Sink.fold(ByteString())(_ ++ _))(Keep.right).run() map { bytes =>
+
```

Comment:
I've stumbled over the same issue in Silhouette. Could you please change the code to something like this:
```scala
      result.body match {
        case body: HttpEntity.Strict =>
          Future.successful(result.copy(
            body = body.copy(
              data = ByteString(compress(body.data))
            )
          ))
        case body: HttpEntity.Streamed =>
          for {
            bytes <- body.data.toMat(Sink.fold(ByteString())(_ ++ _))(Keep.right).run()
          } yield {
            val compressed = compress(bytes)
            val length = compressed.length
            result.copy(
              body = body.copy(
                data = Source.single(ByteString(compressed)),
                contentLength = Some(length.toLong)
              )
            )
          }
        case _ => 
          Future.successful(result)
      }
```

Appending number to variables is a code smell in my eyes :wink: